torchquad/integration/newton_cotes.py

```diff
@@ -208,3 +217,8 @@ def lazy_compiled_integrate(fn, integration_domain):
             return lazy_compiled_integrate
 
         raise ValueError(f"Compilation not implemented for backend {backend}")
+
+    def sum_cur_dim_areas(self, cur_dim_areas, dim, cur_dim):
+        if len(self.integrand_shape) == 0: # i.e it is 1
+               return anp.sum(cur_dim_areas, axis=dim - cur_dim - 1)
+        return anp.sum(cur_dim_areas, axis=len(cur_dim_areas.shape) - 1)
```

Comment:
Are you sure this has to rely on class parameters? To my understanding removing the `staticmethod` above breaks the JIT compilation feature so this will slow it down a lot (iirc). Hence, it would be better if the integrand_shape could be inferred / passed along?

Reply:
I can just move the `integrand_shape` out probably so no issue there.  Good to know.

Reply:
should be fixed!